$DragonFly: src/gnu/usr.bin/cc34/cc_prep/patches/loop.c.patch,v 1.2 2004/06/20 00:10:52 joerg Exp $

diff -u ../gcc.orig/loop.c loop.c
--- ../gcc.orig/loop.c	2004-06-19 22:27:13.000000000 +0200
+++ loop.c	2004-06-20 01:16:04.000000000 +0200
@@ -6513,6 +6513,14 @@
   if (GET_CODE (*mult_val) == USE)
     *mult_val = XEXP (*mult_val, 0);
 
+#ifndef FRAME_GROWS_DOWNWARD
+  if (flag_propolice_protection
+      && GET_CODE (*add_val) == PLUS
+      && (XEXP (*add_val, 0) == frame_pointer_rtx
+	  || XEXP (*add_val, 1) == frame_pointer_rtx))
+    return 0;
+#endif
+
   if (is_addr)
     *pbenefit += address_cost (orig_x, addr_mode) - reg_address_cost;
   else
